packages/grafana-data/src/dataframe/StreamingDataFrame.test.ts

```diff
@@ -214,4 +215,171 @@ describe('Streaming JSON', () => {
     const copy = ({ ...stream } as any) as DataFrame;
     expect(copy.length).toEqual(2);
   });
+
+  describe('streaming labels column', () => {
+    const stream = new StreamingDataFrame(
+      {
+        schema: {
+          fields: [
+            { name: 'time', type: FieldType.time },
+            { name: 'labels', type: FieldType.string },
+            { name: 'speed', type: FieldType.number },
+          ],
+        },
+      },
+      {
+        maxLength: 4,
+      }
+    );
+
+    stream.push({
+      data: {
+        values: [
+          [100, 100],
+          ['sensor=A', 'sensor=B'],
+          [10, 15],
+        ],
+      },
+    });
+
+    stream.push({
+      data: {
+        values: [
+          [200, 200],
+          ['sensor=B', 'sensor=C'],
+          [20, 25],
+        ],
+      },
+    });
+
+    stream.push({
+      data: {
+        values: [
+          [300, 400],
+          ['sensor=A', 'sensor=C'],
+          [30, 40],
+        ],
+      },
+    });
+
+    expect(stream.fields.map((f) => ({ name: f.name, labels: f.labels, values: f.values.buffer })))
+      .toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "labels": undefined,
+          "name": "time",
+          "values": Array [
+            100,
+            200,
+            300,
+            400,
+          ],
+        },
+        Object {
+          "labels": Object {
+            "sensor": "A",
+          },
+          "name": "speed",
+          "values": Array [
+            10,
+            undefined,
+            30,
+            undefined,
+          ],
+        },
+        Object {
+          "labels": Object {
+            "sensor": "B",
+          },
+          "name": "speed",
+          "values": Array [
+            15,
+            20,
+            undefined,
+            undefined,
+          ],
+        },
+        Object {
+          "labels": Object {
+            "sensor": "C",
+          },
+          "name": "speed",
+          "values": Array [
+            undefined,
+            25,
+            undefined,
+            40,
+          ],
+        },
+      ]
+    `);
+  });
+
+  describe('transpose vertical records', () => {
+    let vrecsA = [
+      ['sensor=A', 'sensor=B'],
+      [100, 100],
+      [10, 15],
+    ];
+
+    let vrecsB = [
+      ['sensor=B', 'sensor=C'],
+      [200, 200],
+      [20, 25],
+    ];
+
+    let vrecsC = [
+      ['sensor=A', 'sensor=C'],
+      [300, 400],
+      [30, 40],
+    ];
+
+    let cTables = transpose(vrecsC);
+
+    expect(cTables).toMatchInlineSnapshot(`
+      Array [
+        Array [
+          "sensor=A",
+          "sensor=C",
+        ],
```

Comment:
@ryantxu this might be too cute by half. transpose can return an object instead that has the labels as the keys, e.g.

```js
{
  "sensor=A": [
    [300],
    [30],
  ],
  "sensor=C": [
    [400],
    [40],
  ]
}
```

might make field-matching by label easier.